tests.py

```diff
@@ -1,107 +1,192 @@
+#!/usr/bin/python
+"""Demonstrate Bayesian change-point detection model.
 
-import dateutil,math,numpy,os
+.. codeauthor:: Gabriel Agamennoni <abriel.agamennoni@mavt.ethz.ch>
+.. codeauthor:: Asher Bender <a.bender@acfr.usyd.edu.au>
 
-from dateutil import parser
-from matplotlib import dates,image,pyplot
-from numpy import linalg,random
+"""
+import os
 from os import path
-from scipy import interpolate
 
-from tools import gendata,filterdata,segmentdata,plotprob,plotbound
+import numpy as np
+from numpy import random
+from numpy import linalg
+import matplotlib.pyplot as plt
 
-def synthdata():
+from segmentation import Bcdm
+from segmentation import MatrixVariateNormalInvGamma
 
+
+def random_segments(m, n, k, l, mu=None, omega=None, sigma=None, eta=None,
+                    featfun=None):
+    """Generate random segmented multi-variate linear model data."""
+
+    # Set uninformative prior for the location parameter.
+    if mu is None:
+        mu = np.zeros([m, n])
+
+    # Set uninformative prior for the scale parameter.
+    if omega is None:
+        omega = np.eye(m)
+
+    # Set uninformative prior for the dispersion/noise parameter.
+    if sigma is None:
+        sigma = np.eye(n)
+
+    # Set uninformative prior for the shape parameter.
+    if eta is None:
+        eta = n
```

Comment:
These are not really non-informative priors. They're ambiguous, but somewhat informative. The only way to have a non-informative prior would be to set `omega = epsilon * numpy.eye(m)` and `eta = n + epsilon - 1`, with a really small `epsilon`.